geode-redis/src/main/java/org/apache/geode/redis/internal/PassiveExpirationManager.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.apache.geode.redis.internal;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.geode.logging.internal.executors.LoggingExecutors.newSingleThreadScheduledExecutor;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.CacheClosedException;
+import org.apache.geode.cache.EntryDestroyedException;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.partition.PartitionRegionHelper;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+import org.apache.geode.redis.internal.data.ByteArrayWrapper;
+import org.apache.geode.redis.internal.data.RedisData;
+import org.apache.geode.redis.internal.executor.key.RedisKeyCommands;
+import org.apache.geode.redis.internal.executor.key.RedisKeyCommandsFunctionExecutor;
+
+public class PassiveExpirationManager {
+  private static final Logger logger = LogService.getLogger();
+
+  private final Region<ByteArrayWrapper, RedisData> dataRegion;
+  private final ScheduledExecutorService expirationExecutor;
+
+
+  public PassiveExpirationManager(Region<ByteArrayWrapper, RedisData> dataRegion) {
+    this.dataRegion = dataRegion;
+    expirationExecutor = newSingleThreadScheduledExecutor("GemFireRedis-PassiveExpiration-");
+  }
+
+  public void start() {
+    int INTERVAL = 1;
+    expirationExecutor.scheduleAtFixedRate(() -> doDataExpiration(dataRegion), INTERVAL, INTERVAL,
```

Comment:
I think this would be potentially better with `scheduleWithFixedDelay` so that there's a guaranteed pause between runs. I'm concerned that a large data set would result in `doDataExpiration` taking longer than 1 second and putting pressure on the system by constantly running back-to-back.

Reply:
good idea; it is done